### PR TITLE
Refactoring preprocessing

### PR DESCRIPTION
**Pull request by ErikHartman**
Mostly worked on pathway_network.py. Refactored it to fit new dependencies and better coding practices.

Split binn into binn/model and binn/analysis

Maintained functionality.

Added `reactome_pathway_relation_2025_01_04.txt` and `uniprot_2_reactome_2025_01_14.txt`

### CODE REVIEW

**Comment by ErikHartman** [comment]
Todo:
- Add method to use default reactome pathways

**Comment by ErikHartman** [comment]
Possible add automatic "translation" between reactome ids and pathway names as well as between uniprot and common protein names. What do you think @arnscott

**Comment by arnscott** [comment]
Yea, I think that's a good idea! We have something similar already in DPKS with the annotate() function that will call a uniprot API with https://github.com/multimeric/Unipressed

**Comment by arnscott** [comment]
The only downside with unipressed is that you need an internet connection

**Comment by ErikHartman** [comment]
I ended up refactoring both preprocessing and the model module since these were highly intertwined.

- Made creating a BINN easier. The user now never has to interact with a Network-module or supply their own graphs:
```
# Minimal creation
from binn import BINN
import pandas as pd

input_data = pd.read_csv("../data/test_qm.csv")

binn = BINN(
    data_matrix=input_data,
    use_reactome=True,
    n_layers=4,
)

```
- Created a `BINNHandler` object that deals with preprocessing data and training. Training is now super easy! This took 1.2 seconds on my local Mac (including training!)
```
from binn import BINNHandler

data_handler = BINNHandler(network=binn, save_dir="./logs")

# Align the input data to the network
aligned_data = data_handler.align_to_network(data_matrix=input_data)

# Prepare training data (X, y) from aligned data and design matrix
X, y = data_handler.prepare_training_data(
    data_matrix=aligned_data,
    design_matrix=design_matrix,
    group_column="group",  # Use the column name that specifies groups/classes
    sample_column="sample",  # Use the column name for sample identifiers
)

# Create a PyTorch DataLoader
dataloader = data_handler.create_dataloader(X, y, batch_size=8, shuffle=True)

#  Train the BINN using PyTorch's standard training loop
print("Training BINN with PyTorch...")
data_handler.train_binn_with_pytorch(dataloader, num_epochs=30)

```

**Comment by ErikHartman** [comment]
> Yea, I think that's a good idea! We have something similar already in DPKS with the annotate() function that will call a uniprot API with https://github.com/multimeric/Unipressed

Since this is mostly important during interpretation, I'm saving this for the next stage where I'll refactor SHAP etc.

**Comment by arnscott** [comment]
Nice! Looks clean with the handler

> ```python
> data_handler = BINNHandler(network=binn, save_dir="./logs")
> ```

I wonder if this could be a little clearer, though. Maybe a trainer class can be implemented? Similar to lightning? Something like this instead:

`
binn = BINN(
    data_matrix=input_data,
    use_reactome=True,
    n_layers=4,
)

# Initialize trainer
trainer = BINNTrainer(max_epochs=10)

dataloader = data_handler.create_dataloader(X, y, batch_size=8, shuffle=True)

# Train the model
trainer.fit(binn, dataloader)
`

Quickly reading through your example code, I had to look back and see where the actual BINN was being associated with the loader/handler. I think the common way to handle training is to pass the model in with the `fit()` method. So a small design change might be worth considering to try and match common practice

**Comment by ErikHartman** [comment]
> Nice! Looks clean with the handler
> 
> > ```python
> > data_handler = BINNHandler(network=binn, save_dir="./logs")
> > ```
> 
> I wonder if this could be a little clearer, though. Maybe a trainer class can be implemented? Similar to lightning? Something like this instead:
> 
> ```python
> binn = BINN(
>     data_matrix=input_data,
>     use_reactome=True,
>     n_layers=4,
> )
> 
> # Initialize trainer
> trainer = BINNTrainer(max_epochs=10)
> 
> dataloader = data_handler.create_dataloader(X, y, batch_size=8, shuffle=True)
> 
> # Train the model
> trainer.fit(binn, dataloader)
> ```
> 
> Quickly reading through your example code, I had to look back and see where the actual BINN was being associated with the loader/handler. I think the common way to handle training is to pass the model in with the `fit()` method. So a small design change might be worth considering to try and match common practice

I now split the handler into trainer and dataloader and simplified the dataloader interface:

```py
from binn import BINN, BINNDataLoader, BINNTrainer
import pandas as pd

# Load your data
data_matrix = pd.read_csv("../data/test_qm.csv")
design_matrix = pd.read_csv("../data/design_matrix.tsv", sep="\t")

# Initialize BINN
binn = BINN(data_matrix=data_matrix, use_reactome=True, n_layers=4, dropout=0.2)

## Initialize DataLoader
binn_dataloader = BINNDataLoader(binn)

# Create DataLoaders
dataloaders = binn_dataloader.create_dataloaders(
    data_matrix=data_matrix,
    design_matrix=design_matrix,
    feature_column="Protein",
    group_column="group",
    sample_column="sample",
    batch_size=32,
    validation_split=0.2,
)
# Train the model
trainer = BINNTrainer(binn)
trainer.train(dataloaders=dataloaders, num_epochs=200)

```
